slf4j/src/main/scala-3/org/typelevel/log4cats/slf4j/Slf4jLoggerFactory.scala

```diff
@@ -0,0 +1,48 @@
+package org.typelevel.log4cats.slf4j
+
+import cats.effect.Sync
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+import org.typelevel.log4cats.slf4j.internal.*
+import org.slf4j.Logger as JLogger
+import org.typelevel.log4cats.slf4j.Slf4jLogger.{
+  getLoggerFromClass,
+  getLoggerFromName,
+  getLoggerFromSlf4j
+}
+
+trait Slf4jLoggerFactory[F[_]] {
+  def getLogger(using name: LoggerName): SelfAwareStructuredLogger[F]
+  def getLoggerFromName(name: String): SelfAwareStructuredLogger[F]
+  def getLoggerFromClass(clazz: Class[_]): SelfAwareStructuredLogger[F]
+  def getLoggerFromSlf4j(logger: JLogger): SelfAwareStructuredLogger[F]
+  def create(using name: LoggerName): F[SelfAwareStructuredLogger[F]]
+  def fromName(name: String): F[SelfAwareStructuredLogger[F]]
+  def fromClass(clazz: Class[_]): F[SelfAwareStructuredLogger[F]]
+  def fromSlf4j(logger: JLogger): F[SelfAwareStructuredLogger[F]]
+}
+
+object Slf4jLoggerFactory {
```

Comment:
A little note regarding naming: `cats.effect` uses a way to name factory traits like so:
```scala
trait Ref[F[_]]
//...

object Ref {
  trait Make[F[_]]{
    //...
  } 
}
```
WDYT of using the same naming here for consistency? 






```

Reply:
Those aren't entirely intended for public consumption, not sure if it makes sense to copy that here.

Reply:
Oh, it seems like I got the idea wrong. Could you please show an example of how to use log4cats with slf4j now?

Reply:
Isn't it like that?
```scala
class MyService[F[_]: Monad : Slf4jLoggerFactory] {

val logger = Slf4jLogger.getLogger
//methods
}
```

Reply:
The goals is to be able to use log4cats without needing Sync at the callsite. So using it like you describe in https://github.com/typelevel/log4cats/pull/629#discussion_r834472307 correct.

The instance still needs `Sync`.

Reply:
`Slf4jLoggerFactory[F].getLogger` would be used instead

Reply:
> `Slf4jLoggerFactory[F].getLogger` would be used instead

It's a bit verbose I think. 
Wouldn't it be more convenient to add summoner method right to the Slf4jLogger object like that:
```scala
object Slf4jLogger {
  def getLogger[F[_]: Sync]] = //same as before
 // same methods as before
 def make[F[_]: Slf4jLoggerFactory]: SelfAwareStructuredLogger[F] = Slf4jLoggerFactory[F].getLogger
}
````
This way API changes kinda less.

Reply:
> Those aren't entirely intended for public consumption, not sure if it makes sense to copy that here.

You mean `Ref.Make` or `Slf4jLoggerFactory`?

Reply:
> > Those aren't entirely intended for public consumption, not sure if it makes sense to copy that here.
> 
> You mean `Ref.Make` or `Slf4jLoggerFactory`?

I mean `Ref.Make`

Reply:
`Slf4jLogger.Factory` is what we have now... not sure if I like that.